File: app/src/main/java/com/example/tah/models/BaseViewModel.java
package com.example.tah.models;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import java.util.List;

public abstract class BaseViewModel<T> extends AndroidViewModel {

    protected LiveData<List<T>> itemsLD;
    protected final MutableLiveData<List<Integer>> checkedItemsLD = new MutableLiveData<>();

    public BaseViewModel(@NonNull Application application) {
        super(application);
    }

    public LiveData<List<T>> getItemsLD() {
        return itemsLD;
    }

    public LiveData<List<Integer>> getCheckedItemsLD() {
        return checkedItemsLD;
    }

    public void addToCheckedItems(List<Integer> idList) {
        checkedItemsLD.setValue(idList);
    }

    public abstract void add(T t);
    public abstract void delete(T t);
    public abstract void deleteAll();
}

File: app/src/main/java/com/example/tah/models/BaseViewModel.kt
package com.example.tah.models

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.example.tah.utilities.State
import java.util.*

abstract class BaseViewModel<T>(application: Application)
    : AndroidViewModel(application) {

    lateinit var itemsLD: LiveData<List<T>>
    var checkedItemsLD: MutableLiveData<List<Int>> = MutableLiveData()
    lateinit var state: MutableLiveData<State>


    fun addToCheckedItems(idList: List<Int>){
        checkedItemsLD.value = idList
    }

    abstract fun add(t: T)
    abstract fun delete(t: T)
    abstract fun deleteAll()
    abstract fun deleteSelected()
}